### PR TITLE
Integer mapping for filters (Issue #3)

### PR DESCRIPTION
**Pull request by rlr**
r?

With each commit, I made it more flexible starting from converting all strings no matter what, to mapping types to callables, and finally mapping fields to callables.

### CODE REVIEW

File: oedipus/__init__.py [hunk 1]
```diff
@@ -2,6 +2,7 @@ import logging
 import os
 import re
 import socket
+import zlib
```
Comment (erikrose) [inline oedipus/__init__.py]: Ain't used no mo'.

File: oedipus/__init__.py [hunk 2]
```diff
@@ -187,6 +186,16 @@ class S(elasticutils.S):
                                  'comparator.' %
                                  (comparator, field, comparator, value))
 
+    def _apply_filter_mappings(self, name, value):
+        """Apply filter mappings to convert values to int."""
+        mappings = getattr(self.meta, 'filter_mapping', {})
+        converter = mappings.get(name, None)
+        if converter:
+            if isinstance(value, list):
```
Comment (erikrose) [inline oedipus/__init__.py]: Do we want to support tuples too? We could do `isinstance(value, collections.Iterable)`.

File: oedipus/tests/test_api.py [hunk 3]
```diff
@@ -168,3 +181,13 @@ def test_filter_adapters():
 
 def test_defaults():
     """Defaults from the metadata should be obeyed."""
+
+
+@fudge.patch('sphinxapi.SphinxClient')
+def test_filter_string_mapping(sphinx_client):
```
Comment (erikrose) [inline oedipus/tests/test_api.py]: This is actually the intent of the empty `test_filter_adapters`, so you can kill that. Oh, and can we keep the filled-out tests above the empty ones?

**Comment by erikrose** [comment]
r+ with those little comments. Thanks!

**Comment by rlr** [comment]
3526bed8491177601f26ebd40aecb2377fa062f2

**Pull request closed without merging**